plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -127,21 +156,89 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
-                        if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
-                            actionStatsMap.put(entry.getKey(), entry.getValue());
+            CountDownLatch latch = new CountDownLatch(1);
+            boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+            searchHiddenModels(ActionListener.wrap(hiddenModels -> {
+                for (String modelId : mlStats.getAllModels()) {
+                    if (isSuperAdmin || !hiddenModels.contains(modelId)) {
+                        if (mlStatsInput.retrieveStatsForModel(modelId)) {
+                            Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
+                            for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
+                                if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
+                                    actionStatsMap.put(entry.getKey(), entry.getValue());
+                                }
+                            }
+                            modelStats.put(modelId, new MLModelStats(actionStatsMap));
                         }
                     }
-                    modelStats.put(modelId, new MLModelStats(actionStatsMap));
                 }
+                latch.countDown();
+            }, e -> {
+                log.error("Search Hidden model wasn't successful");
+                latch.countDown();
+            }));
+            // Wait for the asynchronous call to complete
+            try {
+                latch.await(120, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                // Handle interruption if necessary
+                Thread.currentThread().interrupt();
             }
         }
-
         return new MLStatsNodeResponse(clusterService.localNode(), statValues, algorithmStats, modelStats);
     }
+
+    @VisibleForTesting
+    void searchHiddenModels(ActionListener<Set<String>> listener) {
+        SearchRequest searchRequest = buildHiddenModelSearchRequest();
+        // Use a try-with-resources block to ensure resources are properly released
+        try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+            // Wrap the listener to restore thread context before calling it
+            ActionListener<Set<String>> internalListener = ActionListener.runAfter(listener, () -> threadContext.restore());
+            // Wrap the search response handler to handle success and failure cases
+            // Notify the listener of any search failures
+            ActionListener<SearchResponse> al = ActionListener.wrap(response -> {
+                // Initialize the result set
+                Set<String> result = new HashSet<>(response.getHits().getHits().length); // Set initial capacity to the number of hits
```

Comment:
Should we handle edge cases like model index not created yet(for a brand new cluster), model index created but user deleted all models, etc?

Reply:
In my understanding, that should fail in line no 217 and eventually go to the following section:

```
e -> {
                log.error("Search Hidden model wasn't successful");
                latch.countDown();
            }
```

So we aren't showing any model stats finally.

Reply:
questions here, so if the users delete all the models, will the model index still exist? 

I am hoping if there is no models in the cluster, it will delete the model index.

Reply:
The index itself will not be deleted automatically.